lib/msf/core/post/file.rb

```diff
@@ -692,4 +683,43 @@ def _unix_max_line_length
 
     line_max
   end
+
+
+
+  def stat(filename)
+    if session.type == 'meterpreter'
+      return session.fs.file.stat(filename)
+    else
+      raise "`stat' method does not support windows systems" if session.platform == 'windows'
```

Comment:
Could you update this to raise the more specific [`NotImplementedError`](https://ruby-doc.org/core-2.5.0/NotImplementedError.html)?

Reply:
okay :+1: